assert/cmp/compare.go

```diff
@@ -201,12 +202,19 @@ func ErrorContains(err error, substring string) Comparison {
 			return ResultFailure("expected an error, got nil")
 		case !strings.Contains(err.Error(), substring):
 			return ResultFailure(fmt.Sprintf(
-				"expected error to contain %q, got %+v", substring, err))
+				"expected error to contain %q, got %s", substring, formatErrorMessage(err)))
 		}
 		return ResultSuccess
 	}
 }
 
+func formatErrorMessage(err error) string {
+	if errors.Cause(err) == err { // This error was not wrapped with github.com/pkg/errors
```

Comment:
I think this should use a type check:

```go
if _, ok := err.(interface {
    Cause() error
}); ok {
    return ...
}
return ...
```

It's less expensive, seems like it's a bit more obvious, and it will handle weird edge cases where `error` is some other type where `==` doesn't work (ex: struct types or pointers to types that don't compare).

Reply:
Right! It's definitely a better solution 😄